lib/validation.tsx

```diff
@@ -106,26 +108,61 @@ const isFieldResponseValid = (
       break;
     }
     case "fileInput": {
+      //TODO need refactoring.
       const typedValue = value as { file: File; src: FileReader; name: string; size: number };
-      if (validator.required && typedValue.file === null) return t("input-validation.required");
+      if ((validator.required && typedValue === undefined) || typedValue === null)
+        return t("input-validation.required");
+      if (validator.required && typedValue && typedValue.file === null)
+        return t("input-validation.required");
       // Size limit is 8 MB
-      if (typedValue.size > 8000000) return t("input-validation.file-size-too-large");
+      if (typedValue && typedValue.size > 8000000) return t("input-validation.file-size-too-large");
       if (
+        typedValue &&
+        typedValue.file &&
         acceptedFileMimeTypes.split(",").find((value) => value === typedValue.file.type) ===
-        undefined
+          undefined
       ) {
         return t("input-validation.file-type-invalid");
       }
       break;
     }
-    case "dynamicRow": {
-      // Not implemented yet
+    case "dynamicRow":
+      // in the case of the dynamicRow we want to validate the subElements
+      if (formElement.properties.subElements) {
+        // get the subElements from the dynamic row element
+        const subElements = formElement.properties.subElements.filter(
+          (element) => element.type !== "richText"
+        );
+        // set up object to store results
+        const errors: Responses = {};
+        for (const index in subElements) {
+          const subElement = subElements[index];
+          const subElementId = extractSubElementId(subElement);
+          if (subElement.properties.validation && value) {
+            const temValue = value as any;
+            const subElementValues = temValue["0"][parseInt(subElementId)];
```

Comment:
Does this mean we're only validating for the first row only ?

Reply:
No , each element (dynamic row)   has a  array of size 1 that contain all its values.  So this is needed in order to get the subsequent  value or object .

Reply:
@falila 

When I look at the data by consol logging the value that gets passed in to the isFieldResponseValid that does not seem to be the case. In this case here you would only get the first row and validate the subElements only for that row 

![Screen Shot 2021-09-27 at 4 13 05 PM](https://user-images.githubusercontent.com/26819992/134978396-8e6975c4-942d-4a5f-8778-6005c7e90642.png)